modules/core/src/main/java/org/apache/ignite/internal/processors/cache/checker/processor/PartitionReconciliationProcessor.java

```diff
@@ -0,0 +1,600 @@
+
+                if (!recheckKeys.isEmpty())
+                    schedule(
+                        new Recheck(workload.sessionId(), workload.workloadChainId(), recheckKeys,
+                            workload.cacheName(), workload.partitionId(), 0, 0),
+                        recheckDelay,
+                        TimeUnit.SECONDS
+                    );
+            }
+        );
+    }
+
+    /**
+     * @param workload Workload.
+     */
+    private void handle(Recheck workload) throws InterruptedException {
+        compute(
+            CollectPartitionKeysByRecheckRequestTask.class,
+            new RecheckRequest(workload.sessionId(), workload.workloadChainId(), new ArrayList<>(workload.recheckKeys().keySet()), workload.cacheName(),
+                workload.partitionId(), startTopVer),
+            actualKeys -> {
+                Map<KeyCacheObject, Map<UUID, GridCacheVersion>> notResolvingConflicts
+                    = checkConflicts(workload.recheckKeys(), actualKeys,
+                    ignite.cachex(workload.cacheName()).context(), startTopVer);
+
+                if (!notResolvingConflicts.isEmpty()) {
+                    if (workload.recheckAttempt() < recheckAttempts) {
+                        schedule(new Recheck(
+                                workload.sessionId(),
+                                workload.workloadChainId(),
+                                notResolvingConflicts,
+                                workload.cacheName(),
+                                workload.partitionId(),
+                                workload.recheckAttempt() + 1,
+                                workload.repairAttempt()
+                            ),
+                            recheckDelay,
+                            TimeUnit.SECONDS
+                        );
+                    }
+                    else if (fixMode) {
+                        scheduleHighPriority(repair(workload.sessionId(), workload.workloadChainId(), workload.cacheName(), workload.partitionId(), notResolvingConflicts,
+                            actualKeys, workload.repairAttempt()));
+                    }
+                    else {
+                        addToPrintResult(workload.cacheName(), workload.partitionId(), notResolvingConflicts, actualKeys);
+
+                        workProgress.completeWork();
+                    }
+                }
+            });
+    }
+
+    /**
+     * @param workload Workload.
+     */
+    private void handle(Repair workload) throws InterruptedException {
+        compute(
+            RepairRequestTask.class,
+            new RepairRequest(workload.sessionId(), workload.workloadChainId(), workload.data(), workload.cacheName(), workload.partitionId(), startTopVer, repairAlg,
+                workload.repairAttempt()),
+            repairRes -> {
+                if (!repairRes.repairedKeys().isEmpty())
```

Comment:
May be extract body of this function to the method in order to improve readability?